lib/rules/declaration-block-no-redundant-longhand-properties/index.js

```diff
@@ -80,21 +85,69 @@ const rule = (primary, secondaryOptions) => {
 				for (const shorthandProperty of shorthandProperties) {
 					const prefixedShorthandProperty = prefix + shorthandProperty;
 					const longhandDeclaration = longhandDeclarations.get(prefixedShorthandProperty) || [];
+					const longhandDeclarationNode =
+						longhandDeclarationNodes.get(prefixedShorthandProperty) || [];
 
 					longhandDeclaration.push(prop);
 					longhandDeclarations.set(prefixedShorthandProperty, longhandDeclaration);
 
-					const shorthandProps = /** @type {Map<string, Set<string>>} */ (
-						longhandSubPropertiesOfShorthandProperties
-					).get(shorthandProperty);
+					longhandDeclarationNode.push(decl);
+					longhandDeclarationNodes.set(prefixedShorthandProperty, longhandDeclarationNode);
+
+					const shorthandProps =
+						/** @type {import('stylelint').LonghandSubPropertiesOfShorthandProperties} */ (
+							longhandSubPropertiesOfShorthandProperties
+						).get(shorthandProperty);
 					const prefixedShorthandData = Array.from(shorthandProps || []).map(
 						(item) => prefix + item,
 					);
 
-					if (!arrayEqual(prefixedShorthandData.sort(), longhandDeclaration.sort())) {
+					const copiedPrefixedShorthandData = [...prefixedShorthandData];
+
+					if (!arrayEqual(copiedPrefixedShorthandData.sort(), longhandDeclaration.sort())) {
 						continue;
 					}
 
+					if (context.fix) {
+						const declNodes = longhandDeclarationNodes.get(prefixedShorthandProperty) || [];
+
+						if (declNodes !== undefined && declNodes[0] !== undefined) {
```

Comment:
[suggest] We can simplify the code by checking just the first node like this:

```suggestion
						const [firstDeclNode] = declNodes;

						if (firstDeclNode) {
```